modules/apps/transfer/keeper/msg_server.go

```diff
@@ -59,3 +66,32 @@ func (k Keeper) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// unwindHops unwinds the hops present in the tokens denomination and returns the message modified to reflect
+// the unwound path to take. It ensures only a single token must be present in the tokens list
+// and that the token is not native to the chain.
+func (k Keeper) unwindHops(ctx sdk.Context, msg *types.MsgTransfer) (*types.MsgTransfer, error) {
+	coins := msg.GetCoins()
+	token, err := k.tokenFromCoin(ctx, coins[0])
```

Comment:
I know the length has already been verified in ValidateBasic, but I wonder if we should comment or change the godoc because at first glance I was unsure if this actually ensures that a single token is used or just cuts the list if there are multiple - but I guess we working from the assumption that ValidateBasic catches the error of multiple coins.

Reply:
I'll add a comment here for clarity and change the godoc :)